packages/drivers/odsp-driver/src/odspDriverUrlResolver.ts

```diff
@@ -51,31 +52,51 @@ export class OdspDriverUrlResolver implements IUrlResolver {
         return resolveRequest(request);
     }
 
-    public async getAbsoluteUrl(resolvedUrl: IResolvedUrl, relativeUrl: string): Promise<string> {
+    public async getAbsoluteUrl(
+        resolvedUrl: IResolvedUrl,
+        relativeUrl: string,
+        codeDetails?: IFluidCodeDetails,
+    ): Promise<string> {
         let url = relativeUrl;
         if (url.startsWith("/")) {
             url = url.substr(1);
         }
         const odspResolvedUrl = resolvedUrl as IOdspResolvedUrl;
-        return `${odspResolvedUrl.siteUrl}/${url}?driveId=${encodeURIComponent(
-            odspResolvedUrl.driveId)}&itemId=${encodeURIComponent(
-                odspResolvedUrl.itemId,
-            )}&path=${encodeURIComponent("/")}`;
+        let odspUrl = `${odspResolvedUrl.siteUrl}/${url}?driveId=${encodeURIComponent(odspResolvedUrl.driveId,
+            )}&itemId=${encodeURIComponent(odspResolvedUrl.itemId)}&path=${encodeURIComponent("/")}`;
+        const containerPackageName = typeof codeDetails?.package === "string" ? codeDetails.package : undefined;
```

Comment:
you actually want to pull the name off the package. i think it should be something like:

`isFluidPackage(codeDetails?.package) ? codeDetails.package.name : codeDetails.package ?? odspResolvedUrl.codeHint?.containerPackageNeme`

Reply:
That makes sense, so is basically taking the codeDetails into consideration, if not, it takes the codehint. My missing part was the isFluidPackage()